debezium-core/src/test/java/io/debezium/config/ConfigurationTest.java

```diff
@@ -254,4 +254,17 @@ public void testMsgKeyColumnsField() {
         config = Configuration.create().with(MSG_KEY_COLUMNS, "t1,t2").build();
         assertThat(config.validate(Field.setOf(MSG_KEY_COLUMNS)).get(MSG_KEY_COLUMNS.name()).errorMessages()).isNotEmpty();
     }
+
+    @Test
+    @FixFor("DBZ-2957")
+    public void testMsgKeyColumnsFieldRegexValidation() {
+        config = Configuration.create().with(MSG_KEY_COLUMNS, "t1:C1;(.*).t2:C1,C2;").build();
+        assertThat(config.validate(Field.setOf(MSG_KEY_COLUMNS)).get(MSG_KEY_COLUMNS.name()).errorMessages()).isEmpty();
```

Comment:
Perhaps assert the actual message(s)? I.e. which one of of the columns isn't right? That said, I think we still should be lenient towards a trailing semicolon, we can properly interpret the expression also with that. Any other trailing part (e.g. "t1:C1;foobar") would be flagged by your new logic already.